src/components/portal/portal.tsx

```diff
@@ -52,31 +52,45 @@ export const INSERT_POSITIONS: OuiPortalInsertPosition[] = keysOf(
 );
 
 type OuiPortalInsertPosition = keyof typeof insertPositions;
+export type OuiPortalInsert = ExclusiveUnion<
+  { root?: HTMLElement },
+  { sibling: HTMLElement; position: OuiPortalInsertPosition }
+>;
 
 export interface OuiPortalProps {
   /**
    * ReactNode to render as this component's content
    */
   children: ReactNode;
-  insert?: { sibling: HTMLElement; position: OuiPortalInsertPosition };
-  portalRef?: (ref: HTMLDivElement | null) => void;
+  insert?: OuiPortalInsert;
+  portalRef?: (ref: HTMLElement | null) => void;
 }
 
 export class OuiPortal extends Component<OuiPortalProps> {
-  portalNode: HTMLDivElement;
+  portalNode: HTMLElement;
   constructor(props: OuiPortalProps) {
     super(props);
 
     const { insert } = this.props;
 
     this.portalNode = document.createElement('div');
 
+    // no insertion defined, append to body
     if (insert == null) {
-      // no insertion defined, append to body
       document.body.appendChild(this.portalNode);
-    } else {
-      // inserting before or after an element
-      const { sibling, position } = insert;
+      return;
+    }
+
+    const { root, sibling, position } = insert;
+
+    // inserting within an element
+    if (root) {
+      this.portalNode = root;
+      return;
+    }
+
+    // inserting before or after an element
+    if (sibling && position) {
```

Comment:
If we already determine that `root` is not defined, is this if statement necessary? Or is this more of a future-proofing/maintainability decision?

Reply:
Without this condition, we have some typescript errors here, and since we are manipulating the DOM directly, it would be a good idea to check for DOM elements to avoid a sudden crash